.github/workflows/ci.yml

```diff
@@ -179,31 +188,144 @@ jobs:
+
+  # Call reusable workflow
+  call_mapping_reusable_workflow:
+    needs: ['set_up_integration_infra']
+    uses: 'abcxyz/pmap/.github/workflows/snapshot-file-copy.yml@main' # ratchet:exclude
+    with:
+      workload_identity_provider: 'projects/921163060412/locations/global/workloadIdentityPools/github-pool-3b39/providers/github-provider'
+      service_account: 'pmap-3b39-ci-sa@pmap-ci.iam.gserviceaccount.com'
+      destination_prefix: 'pmap-ci/mapping/${{ needs.set_up_integration_infra.outputs.INTEG_TEST_OBJECT_PREFIX_OUTPUT }}'
+      path: 'test/e2e/testdata/fakedata/mapping'
+
+  call_policy_reusable_workflow:
+    needs: ['set_up_integration_infra']
+    uses: 'abcxyz/pmap/.github/workflows/snapshot-file-copy.yml@main' # ratchet:exclude
+    with:
+      workload_identity_provider: 'projects/921163060412/locations/global/workloadIdentityPools/github-pool-3b39/providers/github-provider'
+      service_account: 'pmap-3b39-ci-sa@pmap-ci.iam.gserviceaccount.com'
+      destination_prefix: 'pmap-ci/policy/${{ needs.set_up_integration_infra.outputs.INTEG_TEST_OBJECT_PREFIX_OUTPUT }}'
+      path: 'test/e2e/testdata/fakedata/policy'
+
+  run_integ_test:
+    runs-on: 'ubuntu-latest'
+    needs: ['set_up_integration_infra', 'call_mapping_reusable_workflow', 'call_policy_reusable_workflow']
+    permissions:
+      contents: 'read'
+      id-token: 'write'
+    steps:
+      - uses: 'actions/checkout@ac593985615ec2ede58e132d2e21d2b1cbd6127c' # ratchet:actions/checkout@v3
+      - id: 'auth'
+        uses: 'google-github-actions/auth@ef5d53e30bbcd8d0836f4288f5e50ff3e086997d' # ratchet:google-github-actions/auth@v1
+        with:
+          workload_identity_provider: '${{ env.WIF_PROVIDER }}'
+          service_account: '${{ env.WIF_SERVICE_ACCOUNT }}'
+          token_format: 'access_token'
+      # Write the prefix into GITHUB_ENV which will be used in integration test as a prefix of the
+      # uploaded file's path.
+      - name: 'Write object prefix'
+        run: |
+          echo "INTEG_TEST_OBJECT_PREFIX=${{ needs.set_up_integration_infra.outputs.INTEG_TEST_OBJECT_PREFIX_OUTPUT }}" >> $GITHUB_ENV
+      # Write workflow id to GITHUB_ENV. This variable works as the unique identifier
+      # of the files uploaded by reusable workflow.
+      - name: 'write workflow id to env'
+        id: 'write_workflow_id'
+        run: |
+          echo "WORKFLOW_RUN_ID=${{ github.run_id }}" >> $GITHUB_ENV
       - name: 'Run tests'
         env:
           TEST_INTEGRATION: 'true'
         run: |-
           go test ./test/integration \
             -timeout=15m
+
+  clean_up_integration_infra:
+    runs-on: 'ubuntu-latest'
+    if: '${{ always() }}'
+    needs: ['set_up_integration_infra', 'run_integ_test']
+    permissions:
+      contents: 'read'
+      id-token: 'write'
+    env:
+      MAPPING_TF_VAR_FILE: '${RUNNER_TEMP}/mapping_tf_module/mapping_ci.tfvars'
+      POLICY_TF_VAR_FILE: '${RUNNER_TEMP}/policy_tf_module/policy_ci.tfvars'
```

Comment:
I don't think these files are ever used, they should be empty as they are in a separate job, runner tmp dir is emptied for each job.
Checking the upload and download, the tfvars file is also uploaded and downloaded. I am guessing it is using the tfvar downloaded instead of the one you gave.
<img width="603" alt="Screenshot 2023-06-30 at 11 12 56 AM" src="https://github.com/abcxyz/pmap/assets/51539171/b938d848-4a1a-4a30-9f00-5cd6d15c9c4b">
<img width="770" alt="Screenshot 2023-06-30 at 11 13 52 AM" src="https://github.com/abcxyz/pmap/assets/51539171/255efa83-803e-4157-9a9c-3259297dd893">

Reply:
The `MAPPING_TF_VAR_FILE`, and `POLICY_TF_VAR_FILE` are just the file path, after downloading the artifacts, the files will be in the path, which we use to do destroy.

Reply:
So ${{ runner.temp }} and ${RUNNER_TEMP} is the same thing? (QQ though, if the var files are in the same terraform dir, do we still need to explicit provide tfvar path?)

Reply:
Yes, `${ runner.temp}` is only valid in jobs, can't use in `env` variables. I will update this a bit to get rid of the ${RUNNER_TEMP}.

Reply:
Thanks, just checked, the file has to be exactly terraform.tfvars or terraform.tfvars.json otherwise need to specified that.